pygsti/extras/errorgenpropagation/propagatableerrorgen.py

```diff
@@ -0,0 +1,363 @@
+        errorGens.append( propagatableerrorgen( 'A', [pVec1[1], pVec2[1]] , -1j*wT*pVec1[0]*pVec2[0] ))        
+        pVec1 = labelMultiply(A,Q)
+        pVec2 =labelMultiply(P,B)
+        errorGens.append( propagatableerrorgen( 'A'  , [pVec1[1] , pVec2[1]] , -1j*wT*pVec1[0]*pVec2[0] ))
+        pVec1 = labelMultiply(B,P)
+        pVec2 =labelMultiply(Q,A)                 
+        errorGens.append( propagatableerrorgen( 'A'  , [pVec1[1] , pVec2[1]] , -1j*wT*pVec1[0]*pVec2[0] ))
+        pVec1 = labelMultiply(B,Q)
+        pVec2 =labelMultiply(P,A)
+        errorGens.append( propagatableerrorgen( 'A'  , [pVec1[1] , pVec2[1]] , -1j*wT*pVec1[0]*pVec2[0]))
+        pVec1=acom(A,B)
+        pVec2=com(P,pVec1[1])
+        errorGens.append( propagatableerrorgen( 'A'  , [pVec2[1] , Q ], -.5*1j*wT*pVec1[0]*pVec2[0]))
+        pVec1=acom(A,B)
+        pVec2=com(Q,pVec1[1])
+        errorGens.append( propagatableerrorgen( 'A'  , [pVec2[1], P] , -.5*1j*wT*pVec1[0]*pVec2[0]))
+        pVec1=acom(P,Q)
+        pVec2=com(pVec1[1],A)
+        errorGens.append( propagatableerrorgen( 'A' , [pVec2[1] , B] , -.5*1j*wT*pVec1[0]*pVec2[0]))
+        pVec1=acom(P,Q)
+        pVec2=com(pVec1[1],B)
+        errorGens.append( propagatableerrorgen( 'A' , [pVec2[1] , A ] , -.5*1j*wT*pVec1[0]*pVec2[0]))
+        pVec1=acom(A,B)
+        pVec2=acom(P,Q)
+        pVec3=com(pVec1[1],pVec2[1])
+        errorGens.append( propagatableerrorgen( 'H', [pVec3[1]] ,.25*1j*wT*pVec1[0]*pVec2[0]*pVec3[0]))
+
+    elif ErG1.getType() == 'C' and ErG2.getType() == 'A':
+        A=ErG1.getP1()
+        B=ErG1.getP2()
+        P=ErG2.getP1()
+        Q=ErG2.getP2()
+        pVec1 = labelMultiply(A,P)
+        pVec2 =labelMultiply(Q,B)
+        errorGens.append( propagatableerrorgen('C' , [pVec1[1],pVec2[1]] , 1j*wT*pVec1[0]*pVec2[0]))
+        pVec1 = labelMultiply(A,Q)
+        pVec2 =labelMultiply(P,B)
+        errorGens.append( propagatableerrorgen('C' ,[pVec1[1],pVec2[1]] , -1j*wT*pVec1[0]*pVec2[0]))
+        pVec1 = labelMultiply(B,P)
+        pVec2 =labelMultiply(Q,A)
+        errorGens.append( propagatableerrorgen('C' , [pVec1[1],pVec2[1]] , 1j*wT*pVec1[0]*pVec2[0]))
+        pVec1 = labelMultiply(P,A)
+        pVec2 =labelMultiply(B,Q)
+        errorGens.append( propagatableerrorgen('C'  ,[pVec1[1],pVec2[1]] , -1j*wT*pVec1[0]*pVec2[0]))
+        pVec1 = com(P,Q)
+        pVec2 =com(A,pVec1[1])
+        errorGens.append( propagatableerrorgen('A'  , [pVec2[1] , B] , .5*wT*pVec1[0]*pVec2[0] ))
+        pVec1 = com(P,Q)
+        pVec2 =com(B,pVec1[1])
+        errorGens.append( propagatableerrorgen('A' , [pVec2[1],  A ], .5*wT*pVec1[0]*pVec2[0] ))
+        pVec1 = acom(A,B)
+        pVec2 =com(P,pVec1[1])
+        errorGens.append( propagatableerrorgen('C', [pVec2[1] , Q ], .5*1j*wT*pVec1[0]*pVec2[0] ))
+        pVec1 = acom(A,B)
+        pVec2 =com(Q,pVec1[1])
+        errorGens.append( propagatableerrorgen('C',[pVec2[1],P ],-.5*1j*wT*pVec1[0]*pVec2[0] ))
```

Comment:
Notes have $-\frac{1}{2}$ but @ashenmill confirmed that the coded $-\frac{i}{2}$ is correct.